### PR TITLE
fix for constant float/negative values/references

### PR DESCRIPTION
**Pull request by bekzod**
tried to address https://github.com/emberjs/ember.js/issues/15470

### CODE REVIEW

**Comment by bekzod** [comment]
once merged https://github.com/glimmerjs/glimmer-vm/pull/574 `constants.floats` will store only floats

**Comment by rwjblue** [comment]
Need some help triaging this. The bug it addresses is very serious, but these changes are pretty deep in the details. 

@krisselden / @wycats / @chancancode - can one of y'all review and provide guidance?

**Comment by rwjblue** [comment]
@bekzod - Have you had a chance to look at the CI failures?

**Comment by bekzod** [comment]
@rwjblue not sure why ci is failing :) I think it was passing locally though

**Comment by rwjblue** [comment]
Maybe a transient failure or something that was fixed on CI separately. Can you rebase to see if that does the trick?

**Comment by bekzod** [comment]
done

**Comment by rwjblue** [comment]
@bekzod - The CI failure is type related. `yarn run problems` will allow you to dig into it locally.

**Comment by rwjblue** [comment]
From the Travis logs:

```
$ tsc -p tsconfig.json --noEmit
packages/@glimmer/runtime/lib/compiled/opcodes/vm.ts(5,3): error TS1005: ',' expected.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
The command "yarn run problems" exited with 1.
```

File: packages/@glimmer/runtime/lib/compiled/opcodes/vm.ts [hunk 1]
```diff
@@ -1,15 +1,15 @@
 import { Opaque, Option, SymbolTable } from '@glimmer/interfaces';
-import { ConstReference, Reference, VersionedPathReference } from '@glimmer/reference';
 import {
+  ConstReference,
+  VersionedPathReference
```
Comment (rwjblue) [inline packages/@glimmer/runtime/lib/compiled/opcodes/vm.ts]: Missing a comma here

Comment (bekzod) [inline packages/@glimmer/runtime/lib/compiled/opcodes/vm.ts]: oops, thanks

**Comment by rwjblue** [comment]
@bekzod - Can you rebase?

**Comment by rwjblue** [comment]
@wycats / @chancancode / @krisselden / @chadhietala - We really need to get this bug fixed ASAP, as stated before I am not familiar enough with the internals to know if this is the right direction.  This is a major blocking issue for Ember 2.15's release.

**Comment by bekzod** [comment]
sorry don't have time to resolve conflicts right now, if someone can help out here or I can do it later thanks

**Comment by bekzod** [comment]
rebased

**Comment by krisselden** [comment]
@bekzod thank you for taking care of this 💯

**Pull request merged**